src/memex/search/index.py

```diff
@@ -79,11 +82,35 @@ def delete(es, annotation_id):
 
 
 def reindex(session, es, request):
-    indexing = BatchIndexer(session, es, request)
-    indexing.index_all()
+    """Reindex all annotations into a new index, and update the alias."""
+
+    aliased = es.get_aliased_index() is not None
+
+    # If the index we're using is an alias, then we index into a brand new
+    # index and update the alias at the end.
+    if aliased:
+        new_index = es.index + '-' + _random_id()
+        configure_index(es, index=new_index)
```

Comment:
We can sort this with an [index template](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html) in production.